courseography.cabal

```diff
@@ -50,6 +50,7 @@ test-suite RequirementTests
     test-framework-hunit,
     test-framework-quickcheck2,
     text,
+    time,
```

Comment:
@ChingChang9 everything looks great, I think you just need to remove this dependency (since you've extracted out the timestamps into a separate PR)